src/main/kotlin/anttracker/Request.kt

```diff
@@ -12,92 +12,353 @@ package anttracker.request
 
-    if (contact == null) {
-        println("Error: Bad contact for request")
-        return // bad contact
+    if (displayLeadingBar) {
+        println("| $id | $desc | $priority | $status | $antrel | $created")
+    } else {
+        println("$id | $desc | $priority | $status | $antrel | $created")
     }
+}
 
-    val affRel = request.affectedRelease
-    val date = request.requestDate
-    val name = contact.name
-    val email = contact.email
-    val dept = contact.department
+// ----------------------------------------------------------------
 
-    println("$affRel\t$date\t$name\t$email\t$dept")
+// display the column titles (should precede displayIssue)
+private fun displayIssueColumnTitles(
+    displayLineNum: Boolean // in
+) {
+    // create column strings
+    val linenum = "##".padEnd(3)
+    val id = "ID".padEnd(4)
+    val desc = "Description".padEnd(30)
+    val priority = "Priority".padStart(9)
+    val status = "Status".padEnd(14)
+    val antrel = "AntRel".padEnd(8)
+    val created = "Created".padEnd(10)
+
+    // print columns
+    if (displayLineNum) {
+        println("$linenum | $id | $desc | $priority | $status | $antrel | $created")
+    } else {
+        println("$id | $desc | $priority | $status | $antrel | $created")
+    }
 }
 
+// ----------------------------------------------------------------
+
+/* display list of issues for a given issue, for selecting
+ * issues are ordered by creation date */
+private fun selectIssue(
+    productName: String // in
+): Issue? {
+    // prepare page of issues
+    val issuePage = PageOfIssues(productName)
+    issuePage.loadRecords() // retrieve first page of records
+
+    // selected line number by the user
+    var linenum: Int? = null
+
+    // loop until an issue is selected (linenum)
+    while (linenum == null) {
+        // display page
+        displayIssueColumnTitles(true)
+        issuePage.display()
+
+        // prompt user for input
+        println() // blank line
+        print("Please select issue. ` to create an issue instead: ")
+
+        // switch on user selection
+        when (val selection = readln()) {
+            // abort early - don't select an issue
+            "`" -> return null
+
+            // go to next page
+            "" -> {
+                if (!issuePage.lastPage()) {
+                    issuePage.loadNextPage()
+                } else {
+                    println("No more pages to display.")
+                }
+            }
+
+            // else: check for line # selection
+            else -> {
+                try {
+                    // will throw java.lang.NumberFormatException if not int
+                    val userInputInt = selection.toInt()
+
+                    // validate integer selection
+                    if (issuePage.isValidLineNum(userInputInt)) {
+                        linenum = userInputInt
+                    }
+                } catch (e: java.lang.NumberFormatException) {
```

Comment:
You could match the user's input to a regex to be sure that they are passing you valid input. If not, then you could prompt them again. 

Also, instead of showing the exception message, a custom message would be a better user experience.

Reply:
I'll update the exception message - agree with you there. The range check is consistent with what's being done in other modules, and I think it's pretty clear, I'd like to keep it as is

Reply:
done